public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx

```diff
@@ -343,7 +338,7 @@ export class CloudWatchLogsQueryField extends React.PureComponent<CloudWatchLogs
           <div className="gf-form gf-form--grow flex-shrink-1">
             <QueryField
               additionalPlugins={this.plugins}
-              query={query.expression ?? ''}
+              query={(query as CloudWatchLogsQuery).expression ?? ''}
```

Comment:
can we avoid the cast? (maybe a type guard at the beginning so it's clear that this component uses a `CloudWatchLogsQuery`)

Reply:
yeah good idea. will fix

Reply:
To be able to do this refactoring in a nice way, this component needs to be rewritten from class to function. It's already a bit of a beast with nearly 400 lines of code and lots of state and side effects, so changing it may cause problems that are completely unrelated to annotations. What do you say I create an issue for refactoring this and put it in our backlog? The `query as CloudWatchLogsQuery` is already used in a few places in this component.

Reply:
:thinking: I guess you mean this cast in the state: 

https://github.com/grafana/grafana/blob/5851ce0eb7eab804a2b5984381f1450e9d80e174/public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx#L67

Fine, a follow-up issue sounds good here.

Reply:
Yes and also on line 207. 

New issue: https://github.com/grafana/grafana/issues/48180